tests/dataset_builders/pie/chemprot/test_chemprot.py

```diff
@@ -0,0 +1,1517 @@
+from typing import Union
+
+import datasets
+import pytest
+from pytorch_ie import Document
+from pytorch_ie.documents import (
+    TextDocumentWithLabeledSpansAndBinaryRelations,
+    TextDocumentWithLabeledSpansBinaryRelationsAndLabeledPartitions,
+)
+
+from dataset_builders.pie.chemprot.chemprot import (
+    Chemprot,
+    ChemprotBigbioDocument,
+    ChemprotDocument,
+    example_to_chemprot_bigbio_doc,
+    example_to_chemprot_doc,
+)
+from pie_datasets import load_dataset as load_pie_dataset
+from tests.dataset_builders.common import PIE_BASE_PATH
+
+datasets.disable_caching()
+
+DATASET_NAME = "chemprot"
+BUILDER_CLASS = Chemprot
+PIE_DATASET_PATH = PIE_BASE_PATH / DATASET_NAME
+HF_DATASET_PATH = Chemprot.BASE_DATASET_PATH
+SPLIT_NAMES = {"sample", "test", "train", "validation"}
+SPLIT_SIZES = {"sample": 50, "test": 800, "train": 1020, "validation": 612}
+
+
+@pytest.fixture(scope="module", params=[config.name for config in Chemprot.BUILDER_CONFIGS])
+def dataset_variant(request) -> str:
+    return request.param
+
+
+@pytest.fixture(scope="module")
+def hf_dataset(dataset_variant):
+    return datasets.load_dataset(HF_DATASET_PATH, name=dataset_variant)
+
+
+def test_hf_dataset(hf_dataset):
+    assert hf_dataset is not None
+    assert set(hf_dataset) == {"sample", "train", "validation", "test"}
+    split_sizes = {split_name: len(ds) for split_name, ds in hf_dataset.items()}
+    assert split_sizes == SPLIT_SIZES
+
+
+@pytest.fixture(scope="module")
+def hf_example(hf_dataset):
+    return hf_dataset["sample"][0]
+
+
+def test_hf_example(hf_example, dataset_variant):
```

Comment:
This test is much too long. Shorten it by either:
1. just checking subsets of the data e.g. first and last entries in combination with total amounts of entries, or
2. save the expected data e.g. as json in an appropriate subfolder of `tests/fixtures` and load it from there.

I think I would prefer the first option because then we can still easily see how the HF data looks like, directly from a concise test file (having large json files provides usually not so much insight).

Reply:
Done